module/dkg/client.go

```diff
@@ -181,7 +183,7 @@ func (c *Client) Broadcast(msg model.BroadcastDKGMessage) error {
 // SubmitResult submits the final public result of the DKG protocol. This
 // represents the group public key and the node's local computation of the
 // public keys for each DKG participant. Serialized pub keys are encoded as hex.
-func (c *Client) SubmitResult(groupPublicKey crypto.PublicKey, publicKeys []crypto.PublicKey) error {
+func (c *Client) SubmitResult(groupPublicKey crypto.PublicKey, publicKeys []crypto.PublicKey, indexMap flow.DKGIndexMap) error {
```

Comment:
documentation ⚠️: 
Had a detailed conversation with Tarak this week and he emphasized the following conceptually very important points, which I think would be great to reflect in the documentation
* Conceptually the `flow.DKGIndexMap` is _not_ and output of the DKG protocol. Rather, it is part of the configuration/initialization information of the DKG. 
  *  Hence, I would suggest to rename `SubmitResult` to `SubmitParametersAndResult` and move the input `indexMap` to the front (because DKG configuration happens before DKG results are obtained) 
* In the context of the _happy path_ epoch transition, based on the `EpochSetup` event each consensus participant _locally_ determines DKG committee 𝒟 including the order of the respective nodes order (i.e the information corresponding to `flow.DKGIndexMap`, just not explicitly represented as this struct), 
* However, in case of a failed epoch transition, we desire the ability to manually provide the result of a successful DKG for the immediately next epoch (the so-called recovery epoch). The DKG committee 𝒟 must have a sufficiently _large overlap_ with the recovery epoch's consensus committee 𝒞 -- though for flexibility, we don't want to require that both committees are identical. 

```suggestion
// SubmitParametersAndResult posts the DKG setup parameters (`flow.DKGIndexMap`) and the node's locally-computed DKG result to
// the DKG white-board smart contract. The DKG result are the group public key and the node's local computation of the public
// keys for each DKG participant. Serialized public keys are encoded as hex.
// Conceptually the flow.DKGIndexMap is not and output of the DKG protocol. Rather, it is part of the configuration/initialization
// information of the DKG. Before an epoch transition on the happy path (using the data in the EpochSetup event), each consensus
// participant locally fixes the DKG committee 𝒟 including the order of the respective nodes order to be identical to the consensus
// committee 𝒞. However, in case of a failed epoch transition, we desire the ability to manually provide the result of a successful
// DKG for the immediately next epoch (so-called recovery epoch). The DKG committee 𝒟 must have a sufficiently large overlap with
// the recovery epoch's consensus committee 𝒞 -- though for flexibility, we do *not* want to require that both committees are identical.
// Therefore, we need to explicitly specify the DKG committee 𝒟 on the fallback path. For uniformity of implementation, we do the
// same also on the happy path.
func (c *Client) SubmitParametersAndResult(indexMap flow.DKGIndexMap, groupPublicKey crypto.PublicKey, publicKeys []crypto.PublicKey) error {
```

Note: